sdk/core/azure-core-amqp/src/network/amqp_header_transport.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#include "azure/core/amqp/network/amqp_header_detect_transport.hpp"
+#include "private/transport_impl.hpp"
+
+#include <azure_c_shared_utility/platform.h>
+#include <azure_c_shared_utility/socketio.h>
+#include <azure_uamqp_c/header_detect_io.h>
+#include <azure_uamqp_c/socket_listener.h>
+#include <exception>
+#include <memory>
+#include <stdexcept>
+#include <type_traits>
+
+namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
+
+  AmqpHeaderDetectTransport::AmqpHeaderDetectTransport(
+      std::shared_ptr<Transport> parentTransport,
+      TransportEvents* eventHandler)
+      : Transport(eventHandler)
+  {
+    HEADER_DETECT_IO_CONFIG detectIoConfig{};
+    HEADER_DETECT_ENTRY headerDetectEntries[] = {
+        //        {header_detect_io_get_sasl_amqp_header(), nullptr},
+        {header_detect_io_get_amqp_header(), nullptr},
```

Comment:
Is this a TODO or needs to be removed?

Reply:
Neither - depending on whether you configure your transport for SASL anonymous or not depends on which header detect function should be used.

This is a HORRIBLE design, but it's what we have.